### PR TITLE
fix: wasm module loading

### PR DESCRIPTION
**Pull request by Gozala**
Addresses wasm module loading reported in #175
For details see https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md#the-wasm-unsafe-eval-source-directive

### CODE REVIEW

File: packages/edge-gateway/src/gateway.js [hunk 1]
```diff
@@ -537,7 +537,7 @@ function getTransformedResponseWithCustomHeaders(response) {
 
   clonedResponse.headers.set(
     'content-security-policy',
-    "default-src 'self' 'unsafe-inline' blob: data: ; form-action 'self' ; navigate-to 'self' "
+    "default-src 'self' 'unsafe-inline' 'unsafe-eval' blob: data: ; form-action 'self' ; navigate-to 'self'; "
```
Comment (Gozala) [inline packages/edge-gateway/src/gateway.js:540]:
```suggestion
    "default-src 'self' 'unsafe-inline' 'unsafe-eval' blob: data: ; form-action 'self' ; navigate-to 'self' "
```

File: packages/edge-gateway/src/gateway.js [hunk 2]
```diff
@@ -537,7 +537,7 @@ function getTransformedResponseWithCustomHeaders(response) {
 
   clonedResponse.headers.set(
     'content-security-policy',
-    "default-src 'self' 'unsafe-inline' blob: data: ; form-action 'self' ; navigate-to 'self' "
+    "default-src 'self' 'unsafe-inline' 'unsafe-eval' blob: data: ; form-action 'self' ; navigate-to 'self' "
```
Comment (Gozala) [inline packages/edge-gateway/src/gateway.js:540]: We have considered this in https://github.com/nftstorage/nftstorage.link/pull/172#discussion_r944408575, but left it out. As it turns out leaving it out prevents WASM module loading which is not a desired behavior.

We could enable just wasm via [unsafe-wasm-eval](https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md#the-wasm-unsafe-eval-source-directive) directive, however I don't see a good reason to disable js eval so I'm proposing this.

Comment (Gozala) [inline packages/edge-gateway/src/gateway.js:540]:
```suggestion
    "default-src 'self' 'unsafe-inline' 'unsafe-eval' blob: data: https://*.githubusercontent.com; form-action 'self' ; navigate-to 'self'; connect-src https://polygon-rpc.com "
```

I believe this should address #2143 by enabling github content for any kind of loading and HTTP connections to https://polygon-rpc.com

**Review by hugomrdias — APPROVED** [review]
LGTM

**Review by vasco-santos — APPROVED** [review]
LGTM

File: packages/edge-gateway/src/gateway.js [hunk 3]
```diff
@@ -537,7 +537,7 @@ function getTransformedResponseWithCustomHeaders(response) {
 
   clonedResponse.headers.set(
     'content-security-policy',
-    "default-src 'self' 'unsafe-inline' blob: data: ; form-action 'self' ; navigate-to 'self' "
+    "default-src 'self' 'unsafe-inline' 'unsafe-eval' blob: data: https://*.githubusercontent.com; form-action 'self' ; navigate-to 'self'; connect-src https://polygon-rpc.com "
```
Comment (vasco-santos) [inline packages/edge-gateway/src/gateway.js:540]:
```suggestion
    "default-src 'self' 'unsafe-inline' 'unsafe-eval' blob: data: https://*.githubusercontent.com; form-action 'self' ; navigate-to 'self'; connect-src https://polygon-rpc.com https://rpc.testnet.fantom.network "
```

Looks like we also need fantom per https://github.com/nftstorage/nft.storage/issues/2143#issuecomment-1226617976

**Comment by ikreymer** [comment]
It should probably be: `connect-src 'self' https://polygon-rpc.com https://rpc.testnet.fantom.network`.
Not including `self` ends up blocking loading of local JS within the CID.

**Comment by vasco-santos** [comment]
Thanks @ikreymer ! Yes, it won't inherit from default-src the 'self'. Adding it now

**Pull request merged**